### PR TITLE
Add CLIs `listpeer`, `listtopicpeer`, `removepeer`

### PR DESCRIPTION
**Pull request by mhchia**
### What was wrong?
Currently, we lack of these RPCs

### How was it fixed?
- `listpeer`: list the current peers with connections
- `listtopicpeer topic0 topic1 ...`: list the peers you know who are subscribed to the topics. If there is no topic specified, all topics and its related peers are returned.
- `removepeer peerID`: remove the connection with peer `peerID`

#### Cute Animal Picture

put a cute animal picture link inside the parentheses

### CODE REVIEW

File: requests.go [hunk 1]
```diff
@@ -111,13 +107,9 @@ func (p *RequestProtocol) requestShardPeer(
...
-		peerIDs := []peer.ID{}
-		for _, peerString := range peers.Peers {
-			peerID, err := peer.IDB58Decode(peerString)
-			if err != nil {
-				return nil, fmt.Errorf("error occurred when parsing peerID: %s", peerString)
-			}
-			peerIDs = append(peerIDs, peerID)
+		peerIDs, err := pbPeersToPeerIDs(peers)
+		if err != nil {
+			return nil, fmt.Errorf("peerID in wrong format: %v", err)
```
Comment (NIC619) [inline requests.go]: Nitpick: what do you think about error message: `Failed to convert peer ID from PB string to peerID format, err: %v", err`?

File: rpcserver.go [hunk 2]
```diff
@@ -379,6 +378,63 @@ func (s *server) Send(ctx context.Context, req *pbrpc.SendRequest) (*pbrpc.SendR
...
+	return &pbrpc.RPCListTopicPeerResponse{
+		Response:   makeResponse(true, ""),
+		TopicPeers: topicPeers,
+	}, nil
+}
+
+func (s *server) RemovePeer(
+	ctx context.Context,
+	req *pbrpc.RPCRemovePeerRequest) (*pbrpc.RPCPlainResponse, error) {
+	// Add span for StopServer
```
Comment (NIC619) [inline rpcserver.go]: Should be `RemovePeer`

File: rpcserver.go [hunk 3]
```diff
@@ -379,6 +378,63 @@ func (s *server) Send(ctx context.Context, req *pbrpc.SendRequest) (*pbrpc.SendR
...
+	if err != nil {
+		logger.Debugf("Failed to deserialze the trace context. Tracer won't be able to put rpc call traces together. err: %v", err)
+		spanctx = logger.Start(ctx, "RPCServer.RemovePeer")
+	}
+	defer logger.Finish(spanctx)
+
+	peerID, err := stringToPeerID(req.PeerID)
+	if err != nil {
+		errorMsg := fmt.Sprintf("failed to parse the peerID: %v", req.PeerID)
+		return makePlainResponse(false, errorMsg), nil
```
Comment (NIC619) [inline rpcserver.go]: IMO it makes more sense to `return nil, errorMsg` than `return makePlainResponse(false, errorMsg), nil` when error indeed occurs.

File: rpcserver.go [hunk 4]
```diff
@@ -379,6 +378,63 @@ func (s *server) Send(ctx context.Context, req *pbrpc.SendRequest) (*pbrpc.SendR
...
+	}
+	err = s.node.Network().ClosePeer(peerID)
+	if err != nil {
+		errorMsg := fmt.Sprintf("failed to close the connection to peer: %v", err)
+		return makePlainResponse(false, errorMsg), nil
+	}
+	logger.Debugf("rpcserver:RemovePeer: receive=%v", req)
```
Comment (NIC619) [inline rpcserver.go]: Should we log the message ahead of processing?

File: rpcserver.go [hunk 5]
```diff
@@ -379,6 +378,63 @@ func (s *server) Send(ctx context.Context, req *pbrpc.SendRequest) (*pbrpc.SendR
 	}, nil
 }
 
+func (s *server) ListPeer(
+	ctx context.Context,
+	req *pbrpc.RPCListPeerRequest) (*pbrpc.RPCListPeerResponse, error) {
+	peerIDs := s.node.Network().Peers()
```
Comment (NIC619) [inline rpcserver.go]: Add a logging message for the RPC call?

**Comment by mhchia** [comment]
I think it is ready to be reviewed if the CI passes. Thanks a lot

**Pull request merged**